### PR TITLE
net.waterfox.waterfox

### PR DESCRIPTION
**Pull request by FloridaMan7588**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. I attempted to contact the upstream project owner, but their email address is not public **Link: https://github.com/waterfoxco/waterfox**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). 
     The domain is owned by the aforementioned project owner, and I am unaware of a subdomain delegation policy.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
        The original project does not include an appdata.xml file or a .desktop file for Linux based systems, and doesn't need one in the source project so it won't be upstreamed.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

This is my first time submitting anything here, so apologies to the administrators if this doesn't work out.

### CODE REVIEW

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

File: net.waterfox.waterfox.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - /bin
+      - /etc
+      - /include
+      - /libexec
+      - /share/gtk-doc
+      - /share/man
+  - name: waterfox
+    buildsystem: simple
+    build-commands:
+      - ./install.sh
```
Comment (hfiguiere) [inline net.waterfox.waterfox.yaml:35]: why aren't the build commands here directly?

File: net.waterfox.waterfox.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+app-id: net.waterfox.waterfox
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: waterfox
+finish-args:
+  - "--share=ipc"
+  - "--socket=x11"
+  - "--socket=wayland"
```
Comment (hfiguiere) [inline net.waterfox.waterfox.yaml:9]: does it support wayland or not?

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: that's disappointing.

File: net.waterfox.waterfox.yaml (hunk 1 above)
Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml:35]: It was for testing, need to fig that now

File: net.waterfox.waterfox.yaml (hunk 2 above)
Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml:9]: Left over from doing some testing, should support wayland.

File: flathub.json (hunk 3 above)
Comment (FloridaMan7588) [reply flathub.json:2]: I haven't attempted any cross compiles, I suppose I should leave it up to the bot though.

Comment (FloridaMan7588) [reply flathub.json:2]: After checking, the bin that I'm using doesn't support anything other than x86_64. I did only skim the website though, so unsure if arm64 support exists elsewhere.

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

**Comment by FloridaMan7588** [comment]
fixed those issues, although not the architecture problem. See [here](https://github.com/flathub/flathub/pull/3909#discussion_r1107976815)

File: net.waterfox.waterfox.yaml (hunk 2 above)
Comment (hfiguiere) [reply net.waterfox.waterfox.yaml:8]: then if one has only X11, this will fail.

File: net.waterfox.waterfox.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+finish-args:
+  - "--share=ipc"
+  - "--socket=wayland"
+  - "--filesystem=xdg-download:rw"
+  - "--share=network"
+  - "--socket=pulseaudio"
+  - "--device=all"
+  - "--talk-name=org.freedesktop.Notifications"
+modules:
+  - name: dbus-glib
```
Comment (hfiguiere) [inline net.waterfox.waterfox.yaml:15]: you should ues dbus-glib from the shared-modules.

But is it necessary since you are not building from source?

Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml:15]: Will switch to dbus-glib, and yes it is necessary unfortunately, it throws XPCOM errors without dbus-glib

File: net.waterfox.waterfox.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    buildsystem: simple
+    build-commands:
+      - mkdir /app/opt
+      - cp -r waterfox/ /app/opt/waterfox
+      - desktop-file-install --dir /app/share/applications net.waterfox.waterfox.desktop
+      - mkdir -p /app/share/icons/hicolor/256x256/apps/ /app/share/appdata
+      - cp -p /app/opt/waterfox/browser/chrome/icons/default/default256.png /app/share/icons/hicolor/256x256/apps/net.waterfox.waterfox.png
+      - cp -p net.waterfox.waterfox.appdata.xml /app/share/appdata
+      - mkdir -p /app/opt/waterfox/browser/defaults/preferences
+      - cp -p flatpak-prefs.js /app/opt/waterfox/browser/defaults/preferences/
```
Comment (hfiguiere) [inline net.waterfox.waterfox.yaml:41]:
```suggestion
      - install -Dm644 /app/opt/waterfox/browser/chrome/icons/default/default256.png /app/share/icons/hicolor/256x256/apps/net.waterfox.waterfox.png
      - install -Dm644 net.waterfox.waterfox.appdata.xml -t /app/share/appdata
      - install -Dm644 flatpak-prefs.js -t /app/opt/waterfox/browser/defaults/preferences/
```

Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml]: Added

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

**Comment by barthalion** [comment]
> The domain is owned by the aforementioned project owner, and I am unaware of a subdomain delegation policy.

Have you contacted the developers about this submission?

**Comment by FloridaMan7588** [comment]
The developer has no public email address, should I open an issue on the project?

**Comment by hfiguiere** [comment]
That's how one usually go about it.

**Comment by FloridaMan7588** [comment]
On it!

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

File: net.waterfox.waterfox.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+app-id: net.waterfox.waterfox
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: waterfox
+finish-args:
+  - "--share=ipc"
+  - "--socket=wayland"
+  - "--filesystem=xdg-download:rw"
+  - "--filesystem=~/.waterfox:create"
```
Comment (hfiguiere) [inline net.waterfox.waterfox.yaml:10]: You should use --persist instead.

Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml:10]: I've switched my test build to use --persist, and it works just fine, but where does it store the data?

Comment (FloridaMan7588) [reply net.waterfox.waterfox.yaml]: I found the data location.

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

**Comment by FloridaMan7588** [comment]
bot, build net.waterfox.waterfox

File: net.waterfox.waterfox.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    build-commands:
+      - mkdir /app/opt
+      - cp -r waterfox/ /app/opt/waterfox
+      - desktop-file-install --dir /app/share/applications net.waterfox.waterfox.desktop
+      - install -Dm644 /app/opt/waterfox/browser/chrome/icons/default/default256.png /app/share/icons/hicolor/256x256/apps/net.waterfox.waterfox.png
+      - install -Dm644 net.waterfox.waterfox.appdata.xml -t /app/share/appdata
+      - install -Dm644 flatpak-prefs.js -t /app/opt/waterfox/browser/defaults/preferences/
+      - ln -sf /app/opt/waterfox/waterfox-bin /app/bin/waterfox
+    sources:
+      - type: archive
```
Comment (barthalion) [inline net.waterfox.waterfox.yaml:31]: Can you configure external-data-checker for that module? https://github.com/flathub/flatpak-external-data-checker/

**Comment by barthalion** [comment]
bot, build net.waterfox.waterfox

**Pull request closed without merging**